### PR TITLE
feat(gatsby-source-contentful): Give "Asset" the same Sys fields as "Entry"

### PR DESCRIPTION
**Pull request by gburgett**
## Description

Howdy all!

In my Gatsby project, I have a need to create a reproducible ETag & Last Modified of a tree of Contentful entries.  To generate my last-modified date, I take the max of all the `updatedAt` timestamps of all the entries in my tree.  Unfortunately I can't yet include Assets in my tree, so when asset descriptions change I'm forced to republish the root element that links to the asset in order to update the last-modified.

I propose to add the same fields that exist on the "Entry" node to "Asset" as well.  I am including only the fields that the linked documentation says are available to both Entries and Assets.

Appreciate your hard work and the product!

### Documentation

https://www.contentful.com/developers/docs/references/content-delivery-api/#/introduction/common-resource-attributes

### CODE REVIEW

File: packages/gatsby-source-contentful/src/normalize.js [hunk 1]
```diff
@@ -623,6 +626,12 @@ exports.createAssetNodes = ({
       internal: {
         type: `${makeTypeName(`Asset`)}`,
       },
+      sys: {}
```
Comment (axe312ger) [inline packages/gatsby-source-contentful/src/normalize.js:629]: Why the empty sys?

File: packages/gatsby-source-contentful/src/normalize.js [hunk 2]
```diff
@@ -623,6 +626,12 @@ exports.createAssetNodes = ({
...
+    }
+    
+        // Revision applies to entries, assets, and content types
+    if (assetItem.sys.revision) {
```
Comment (axe312ger) [inline packages/gatsby-source-contentful/src/normalize.js:633]: How can this be true? 🤔

Comment (axe312ger) [reply packages/gatsby-source-contentful/src/normalize.js:633]: Ohh sorry,  `assetItem` !== `assetNode`


But we should move revision out of sys as we extract all properties from sys into the root of the node above.

Comment (gburgett) [reply packages/gatsby-source-contentful/src/normalize.js:633]: I agree, however I copied this code from above to be consistent with what is already defined for Entry nodes.  If we went back and changed it for Entry nodes, that would be a breaking change.

In my personal opinion, I think they should be in both places.  If someone defines a content type that has a text field named `revision`, then that would (appropriately) override the `revision` attribute that came from sys.  I think that is good behavior, but I would also like to be able to access it via `{ sys { revision } }` in that case.

However, the above is just a hypothetical.  We can move it to the root and just see if anyone has that problem and wants to complain :)

Comment (axe312ger) [reply packages/gatsby-source-contentful/src/normalize.js:633]: Alright thanks! We will keep it this way in this PR, later on in a breaking release we gonna change the structures of entries and assets together.

**Comment by gburgett** [comment]
Sure thing!  I'm sorry this totally slipped off my radar this past week

**Comment by gburgett** [comment]
@axe312ger would you help me with the integration tests?  There appears to be a permissions issue on CircleCI.

**Comment by axe312ger** [comment]
@gburgett Hey, just had a look and looks like a strange permission error. Could you please rebase on upstream/master and push? That should/hopefully will fix it

Thanks :)

**Comment by gburgett** [comment]
@axe312ger It looks like rebasing solved those issues!  All checks have passed now.

**Comment by gburgett** [comment]
Hi @axe312ger , any update on when we can expect this to be merged and released?

**Comment by axe312ger** [comment]
I'll notify the core team about the missing review :)

**Comment by gburgett** [comment]
Hi @axe312ger or @wardpeet, where can I go to check if this has been released?

**Comment by axe312ger** [comment]
The change log for the plugin is here:

https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-contentful/CHANGELOG.md

Was released in 2.3.27 (2020-07-17)

**Pull request merged**